### PR TITLE
fix index catchup to work on start

### PR DESCRIPTION
**Pull request by vdaniloff**
1) Only apply the stream id check during normal operations, not startup replay
2) use the recording id on startup, not the aeron session id to lookup the recording
3) change indexedpositionwriter to use the recording id and not the session id
4) initialise the record offsets map on loading file.

found no easy way to find recording by replay session id

### CODE REVIEW

**Comment by vdaniloff** [comment]
looks like a test fails, will continue

File: artio-codecs/src/test/java/uk/co/real_logic/artio/fields/DecimalFloatToAndFromDoubleTest.java [hunk 1]
```diff
@@ -31,40 +32,39 @@ public void shouldExtractNaNFromDecimalNumber()
     @Test
     public void shouldConvertFromDoubleAccurately()
     {
-        final StringBuilder builder = new StringBuilder();
```
Comment (RichardWarburton) [inline artio-codecs/src/test/java/uk/co/real_logic/artio/fields/DecimalFloatToAndFromDoubleTest.java]: I don't think these decimal float changes look related to the indexer catchup, can you keep them out of the PR please.

Comment (vdaniloff) [reply artio-codecs/src/test/java/uk/co/real_logic/artio/fields/DecimalFloatToAndFromDoubleTest.java]: removed. In Russia this test fails :)

File: artio-core/src/main/java/uk/co/real_logic/artio/engine/PositionSender.java [hunk 2]
```diff
@@ -96,4 +96,11 @@ public void close()
     public void readLastPosition(final IndexedPositionConsumer consumer)
     {
     }
+
+    @Override
```
Comment (RichardWarburton) [inline artio-core/src/main/java/uk/co/real_logic/artio/engine/PositionSender.java]: We don't use @Override in real logic projects.

File: artio-core/src/main/java/uk/co/real_logic/artio/engine/framer/FramerContext.java [hunk 3]
```diff
@@ -86,9 +86,11 @@ public FramerContext(
             outboundLibraryStreams.dataPublication("outboundPublication"));
 
         sentSequenceNumberIndex = new SequenceNumberIndexReader(
-            configuration.sentSequenceNumberBuffer(), errorHandler, configuration.logFileDir());
+            configuration.sentSequenceNumberBuffer(), errorHandler, recordingCoordinator.outboundRecordingIdLookup(),
```
Comment (RichardWarburton) [inline artio-core/src/main/java/uk/co/real_logic/artio/engine/framer/FramerContext.java]: These changes now mean that the RecordingIdLookup class can be used on the Framer and Indexer agents. It's not a class that is designed to be threadsafe. Probably best to pass out separate instances to each thread.

**Comment by RichardWarburton** [comment]
Thanks for the PR @vdaniloff - My only other suggestion is that this PR doesn't have a way to reproduce the original bug.

File: artio-core/src/main/java/uk/co/real_logic/artio/engine/PositionSender.java (hunk 2 above)
Comment (vdaniloff) [reply artio-core/src/main/java/uk/co/real_logic/artio/engine/PositionSender.java]: removed

File: artio-core/src/main/java/uk/co/real_logic/artio/engine/framer/FramerContext.java (hunk 3 above)
Comment (vdaniloff) [reply artio-core/src/main/java/uk/co/real_logic/artio/engine/framer/FramerContext.java]: fixed, thanks!

**Comment by vdaniloff** [comment]
I suspect this might be the reason PersistentSequenceNumberResentRequestSystemTest fails with "shutdownClearly=false", currently struggling with getting it passed on Windows. 

Originally I was more concerned with repeating reload in "extendRecording" case, since there would be multiple replays for same recording (some of that stored by old aeron session id - so potentially huge)

**Comment by vdaniloff** [comment]
I temporary admit my defeat with PersistentSequenceNumberResentRequestTest, and normal graceful shutdown seems to have no functional impact, except for performance in extendRecording case. The rest is fixed.

**Pull request merged**